zetaclient/orchestrator/v2_bootstrap.go

```diff
@@ -131,6 +136,57 @@ func (oc *V2) bootstrapEVM(ctx context.Context, chain zctx.Chain) (*evm.EVM, err
 	return evm.New(oc.scheduler, observer, signer), nil
 }
 
+func (oc *V2) bootstrapSolana(ctx context.Context, chain zctx.Chain) (*solana.Solana, error) {
+	// should not happen
+	if !chain.IsSolana() {
+		return nil, errors.New("chain is not Solana")
+	}
+
+	app, err := zctx.FromContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	cfg, found := app.Config().GetSolanaConfig()
+	if !found {
+		return nil, errors.Wrap(errSkipChain, "unable to find solana config")
+	}
+
+	baseObserver, err := oc.newBaseObserver(chain, chain.Name())
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to create base observer")
+	}
+
+	gwAddress := chain.Params().GatewayAddress
+
+	rpcClient := solrpc.New(cfg.Endpoint)
+	if rpcClient == nil {
+		return nil, errors.New("unable to create rpc client")
+	}
+
+	observer, err := solbserver.New(baseObserver, rpcClient, gwAddress)
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to create observer")
+	}
+
+	// try loading Solana relayer key if present
+	password := chain.RelayerKeyPassword()
+	relayerKey, err := keys.LoadRelayerKey(app.Config().GetRelayerKeyPath(), chain.RawChain().Network, password)
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to load relayer key")
+	}
```

Comment:
while it may look like this will fail if the relayer key is not present, it will just `return nil, nil` if unset